docs/developer-docs/integrations/sns/managing/making-proposals.md

```diff
@@ -42,6 +42,619 @@ For these cases, SNSs have so called 'generic proposals'. These are custom propo
+                wait_for_quiet_deadline_increase_seconds = opt 42:nat64;
+                max_number_of_neurons = opt 42:nat64;
+                transaction_fee_e8s = opt 42:nat64;
+                max_number_of_proposals_with_ballots = opt 42:nat64;
+                max_age_bonus_percentage = opt 42:nat64;
+                neuron_grantable_permissions = opt record { permissions : vec {1:int32, 2:int32} };
+                voting_rewards_parameters = opt record {
+                    final_reward_rate_basis_points = opt 42:nat64;
+                    initial_reward_rate_basis_points = opt 42:nat64;
+                    reward_rate_transition_duration_seconds = opt 42:nat64;
+                    round_duration_seconds = opt 42:nat64;
+                };
+                maturity_modulation_disabled = opt true:bool;
+                max_number_of_principals_per_neuron = opt 42:nat64;
+            };
+    }
+)'
+```
+
+## `AddGenericNervousSystemFunction`
+
+### Relevant type signatures
+
+```candid
+    type AddGenericNervousSystemFunction : NervousSystemFunction;
+
+    type NervousSystemFunction = record {
+        id : nat64;
+        name : text;
+        description : opt text;
+        function_type : opt FunctionType;
+    };
+
+    type FunctionType = variant {
+        NativeNervousSystemFunction : record {};
+        GenericNervousSystemFunction : GenericNervousSystemFunction;
+    };
+
+    type GenericNervousSystemFunction = record {
+        validator_canister_id : opt principal;
+        target_canister_id : opt principal;
+        validator_method_name : opt text;
+        target_method_name : opt text;
+    };
+```
+
+### Putting it together
+
+```candid
+    type AddGenericNervousSystemFunction: record {
+        id : nat64;
+        name : text;
+        description : opt text;
+        function_type : opt variant {
+            NativeNervousSystemFunction : record {};
+            GenericNervousSystemFunction : record {
+                validator_canister_id : opt principal;
+                target_canister_id : opt principal;
+                validator_method_name : opt text;
+                target_method_name : opt text;
+            }
+        };
+    };
+```
+
+Example in bash:
+
+```bash
+quill sns make-proposal <PROPOSER_NEURON_ID> --proposal '(
+    record {
+        title = "SNS Launch";
+        url = "https://dfinity.org";
+        summary = "A motion to start the SNS";
+        action = opt variant {
+            AddGenericNervousSystemFunction = record {
+                id = nat64;
+                name = text;
+                description = opt text;
+                function_type : opt variant {
+                    NativeNervousSystemFunction = record {};
+                    GenericNervousSystemFunction = record {
+                        validator_canister_id = opt principal "ltyfs-qiaab-aaaak-aan3a-cai";
+                        target_canister_id = opt principal;
```

Comment:
I like that. I would fill in the same principal for the target_canister_id and then adjust the summary above accordingly to have one concrete example

Reply:
done